src/FESpaces/FESpacesWithDofPotentiallyRemoved.jl

```diff
@@ -0,0 +1,185 @@
+"""
+    struct FESpaceWithDofPotentiallyRemoved{CS,RD} <: SingleFieldFESpace
+      space::SingleFieldFESpace
+      constraint_style::Val{CS}
+      remove_dof::Val{RD}
+      dof_to_remove::Int
+    end
+"""
+struct FESpaceWithDofPotentiallyRemoved{CS,RD} <: SingleFieldFESpace
+  space::SingleFieldFESpace
+  constraint_style::Val{CS}
+  remove_dof::Val{RD}
+  dof_to_remove::Int
+  @doc """
+      FESpaceWithDofPotentiallyRemoved(space::SingleFieldFESpace, remove_dof::Bool,
+      dof_to_remove::Int=num_free_dofs(space))
+  """
+  function FESpaceWithDofPotentiallyRemoved(space::SingleFieldFESpace,
+     remove_dof::Bool,
+     dof_to_remove::Int=num_free_dofs(space))
+    s = "FESpaceWithDofPotentiallyRemoved can only be constructed from spaces without dirichlet dofs."
+    @notimplementedif num_dirichlet_dofs(space) != 0 s
+    @assert !remove_dof || 1 <= dof_to_remove <= num_free_dofs(space)
+    cs = constraint_style(space)
+    CS = get_val_parameter(cs)
+    rd = Val{remove_dof}()
+    RD = remove_dof
+    new{CS,RD}(space,cs,rd,dof_to_remove)
+  end
+end
+
+# Genuine functions
+function num_free_dofs(f::FESpaceWithDofPotentiallyRemoved{CS,true}) where {CS}
+  num_free_dofs(f.space)-1
+end
+
+function num_free_dofs(f::FESpaceWithDofPotentiallyRemoved{CS,false}) where {CS}
+  num_free_dofs(f.space)
+end
+
+function zero_free_values(f::FESpaceWithDofPotentiallyRemoved)
+  zeros(num_free_dofs(f))
+end
+
+function get_cell_dofs(f::FESpaceWithDofPotentiallyRemoved{CS,true}) where {CS}
+  cell_dofs = get_cell_dofs(f.space)
+  CellDofsWithDofRemoved(cell_dofs,f.dof_to_remove)
+end
+
+function get_cell_dofs(f::FESpaceWithDofPotentiallyRemoved{CS,false}) where {CS}
+  get_cell_dofs(f.space)
+end
+
+
+num_dirichlet_dofs(f::FESpaceWithDofPotentiallyRemoved{CS,true}) where {CS} = 1
+
+num_dirichlet_dofs(f::FESpaceWithDofPotentiallyRemoved{CS,false}) where {CS} = 0
+
+function zero_dirichlet_values(f::FESpaceWithDofPotentiallyRemoved)
+  T = Float64 # TODO
+  zeros(T,num_dirichlet_dofs(f))
+end
+
+
+num_dirichlet_tags(f::FESpaceWithDofPotentiallyRemoved{CS,true}) where {CS} = 1
+
+num_dirichlet_tags(f::FESpaceWithDofPotentiallyRemoved{CS,false}) where {CS} = 0
+
+get_dirichlet_dof_tag(f::FESpaceWithDofPotentiallyRemoved{CS,true}) where {CS} = Int8[1,]
+
+get_dirichlet_dof_tag(f::FESpaceWithDofPotentiallyRemoved{CS,false}) where {CS} = Int8[]
+
+function scatter_free_and_dirichlet_values(
+  f::FESpaceWithDofPotentiallyRemoved{CS,true},fv,dv) where {CS}
+  @assert length(dv) == 1
+  _dv = similar(dv,eltype(dv),0)
+  _fv = vcat(view(fv,1:f.dof_to_remove-1),
+             dv,
+             view(fv,f.dof_to_remove:length(fv))) # TODO lazy append
```

Comment:
Try `Gridap.Arrays.lazy_append`

Reply:
The other option is to implement a `VectorWithIndexInserted`, or whatever name you like.